### PR TITLE
Search input

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: assets/js/main.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
+import {cities} from "./list-of-cities.js";
+
+const searchInput = document.getElementById("search-input");
+
 const cityDropdown = document.getElementById("cities");
```
Comment (blacksrc) [inline assets/js/main.js]: This not dropdown anymore.

File: assets/js/main.js [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
 }
 
-function createCitiesOptions(cities) {
-  let optionsHtml = "";
-  cities.forEach((city) => {
-    optionsHtml += `<option value="${generateValueFromCity(city)}">${
-      city.name
-    }</option>`;
+searchInput.addEventListener("keyup", function() {
+  console.log(filterCitiesByName(searchInput.value.toLowerCase(), cities))
```
Comment (blacksrc) [inline assets/js/main.js]: please implement the expected behavior here instead of `console.log`

Comment (blacksrc) [reply assets/js/main.js]: This probably should show the filtered result to the end user. You can use datalist to achieve that:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

File: assets/js/main.js [hunk 3]
```diff
@@ -1,24 +1,23 @@
...
-  { name: "Rasht", countryCode: "IR" },
-  { name: "Tehran", countryCode: "IR" },
-  { name: "Stockholm", countryCode: "SE" },
-];
-
 function generateValueFromCity(city) {
   return `${city.name},${city.countryCode}`;
 }
 
-function createCitiesOptions(cities) {
```
Comment (blacksrc) [inline assets/js/main.js:14]: You probably need this function back but with a bit of modification.

File: assets/js/main.js [hunk 4]
```diff
@@ -118,9 +117,11 @@ async function displayWeather(selectedCity) {
   showCityWeather(city, list);
 }
 
-cityDropdown.addEventListener("change", async () => {
-  displayWeather(cityDropdown.value);
+searchInput.addEventListener('keyup', function (e) {
+  if (e.key === 'Enter') {
+    displayWeather(searchInput.value);
+    searchInput.value= "";
```
Comment (blacksrc) [inline assets/js/main.js:123]: There is no need to clear up the input.

File: index.html [hunk 5]
```diff
@@ -11,9 +10,9 @@
   </head>
   <body>
     <div class="site-wrapper">
-      <input type="search" list="cities" />
+      <input id="search-input" type="search" list="cities" />
```
Comment (blacksrc) [inline index.html]: nice naming.

File: assets/js/main.js (hunk 2 above)
Comment (blacksrc) [reply assets/js/main.js]: Move the `.toLowerCase()` part into the function.

File: index.html (hunk 5 above)
Comment (hedita) [reply index.html]: Thanks.

File: assets/js/main.js (hunk 1 above)
Comment (hedita) [reply assets/js/main.js]: Done.

File: assets/js/main.js (hunk 2 above)
Comment (hedita) [reply assets/js/main.js]: Done.

**Review by blacksrc — APPROVED** [review]

**Pull request merged**